src/Database/Bloodhound/Raw.hs

```diff
@@ -129,77 +76,67 @@ search index body scrollRequest = do
 -- | A special purpose search implementation that uses the faster json-syntax
 searchHit ::
   MonadBH m =>
-  Aeson.ToJSON body =>
+  Aeson.FromJSON a =>
   BH.IndexName ->
-  body ->
-  m [Json.Value]
```

Comment:
Unfortunately, aeson is too slow for certain query, so we do need to keep json-syntax here. See: the test/JsonDecode.{hs,py} benchmark.